active/Win32/Window_and_Application_API_Spec.md

```diff
@@ -0,0 +1,618 @@
+
+### Window.CoreWindow property: 
+
+Get the CoreWindow of a Window when the Window was created in a UWP app model.
+
+```CS
+public CoreWindow CoreWindow { get; }
+```
+
+#### Remarks
+- **Win32**: This property returns null when running in Win32 app model. 
+
+### IWindowInterop Interface 
+Exposes a COM interface (only callable from Win32) to get the HWnd of a Window in a Win32 app model.
+
+```CS
+Window w = new Window();
+w.QueryInterface(IWindowInterop);
+IntPtr hwnd = w.GetInterop<IWindowInterop>().Handle;
+```
+
+### **Others**
+### Window.Dispatcher property
+Gets the Dispatcher object for the Window.
+
+```CS
+public CoreDispatcher Dispatcher { get; }
+```
+#### OPEN ISSUE: 
+We are Still figuring out what type return. There is no CoreDispatcher in Win32 (DisptacherQueue works) and the current UWP behavior gets the CoreDispatcher object for the Window.
+
+Evaluating: 
+1. Create a 'XamlDispatcher' that uses underneath the available dispatcher.
+2. Make CoreDispatcher works in Win32
+3. Returns DispatcherQueue in Win32 and UWP
+4. Returns null in Win32 and keep using CoreDispatcher. Win32 devs should use DisptacherQueue.GetForCurrentThread()
+
+### Window.UIContext property **[Remove it]**
+Gets the context identifier for the window. Basically, this is used by InputPane and CoreInputView APIs to figure out where to display the Software Input Panel (SIP). 
+
+```CS
+public UIContext UIContext {get;}
+```
+
+#### Decision:
+We should do here whatever we decide to do on UIElement. The current plan is to remove it.
+
+### Window.Compositor property
+
+ Gets the Compositor for this Window. This allows to access to the Visual Layer (the layer where the XAML framework sits) and use WinRT compositor APIs inside of the Window. 
+
+```CS
+public Microsoft.UI.Composition.Compositor Compositor { get; }
+```
+### Remarks
+There is one Compositor per thread, that means that all the Windows in the same thread share the same Compositor.
+
+
+
+## WindowEventArgs class
+Namespace: Microsoft.UI.Xaml
+
+Contains the set of arguments returned to an app after a window input or behavior event.
+
+### Properties
+
+- **WindowEventArgs.Handled**: Gets or sets whether the event was handled. **true** if the event has been handled by the appropriate delegate; **false** if it has not.
+    ```CS
+    public bool Handled { get; set; }
+    ```
+
+
+## WindowActivatedEventArgs class
+Namespace: Microsoft.UI.Xaml
+
+Contains the windows activation state information returned by the Window.Activated event.
+
+```CS
+public class WindowActivatedEventArgs
+```
+
+### Properties
+- **WindowActivatedEventArgs.Handled**: Gets or sets whether the window activation changed event was handled. **true** if the event has been handled by the appropriate delegate; **false** if it has not.
+    ```CS
+    public bool Handled { get; set; }
+    ```
```

Comment:
What's the point of handling this? Will that prevent it from getting activated? If I don't handle it, what will it bubble up to?

Reply:
We kept it for backwards compatibility with the existing APIs. Honestly, I didn't find a scenario that justify this, but perhaps my scope (from XAML) is narrower than the scope of UWP windowing creators.